src/test/java/com/conveyal/datatools/manager/controllers/api/FeedSourceControllerTest.java

```diff
@@ -104,10 +133,71 @@ public void createFeedSourceWithNoUrlTest() {
         assertEquals(0, jobCountForFeed(feedSourceWithNoUrl.id));
     }
 
+
+    /**
+     * Create some labels, add them to the feed source make them admin only, and check that they don't appear if not an admin
+     */
+    @Test
+    public void createFeedSourceWithLabels() throws IOException {
+        // Create labels
+        HttpResponse createFirstLabelResponse = TestUtils.makeRequest("/api/manager/secure/label",
+                JsonUtil.toJson(publicLabel),
+                HttpUtils.REQUEST_METHOD.POST
+        );
+        assertEquals(OK_200, createFirstLabelResponse.getStatusLine().getStatusCode());
+        HttpResponse createSecondLabelResponse = TestUtils.makeRequest("/api/manager/secure/label",
+                JsonUtil.toJson(adminOnlyLabel),
+                HttpUtils.REQUEST_METHOD.POST
+        );
+        assertEquals(OK_200, createSecondLabelResponse.getStatusLine().getStatusCode());
+
+        String firstLabelId = publicLabel.id;
+        String secondLabelId = adminOnlyLabel.id;
+
+        feedSourceWithLabels.labelIds.add(firstLabelId);
+        feedSourceWithLabels.labelIds.add(secondLabelId);
+
+        // Create feed source with labels
+        HttpResponse createFeedSourceResponse = TestUtils.makeRequest("/api/manager/secure/feedsource",
+                JsonUtil.toJson(feedSourceWithLabels),
+                HttpUtils.REQUEST_METHOD.POST
+        );
+        assertEquals(OK_200, createFeedSourceResponse.getStatusLine().getStatusCode());
+
+        // Test that they are assigned properly
+        assertEquals(2, labelCountForFeed(feedSourceWithLabels.id));
+        // Test that project shows only correct labels based on user auth
+        assertEquals(2, labelCountforProject(feedSourceWithLabels.projectId, true));
+        assertEquals(1, labelCountforProject(feedSourceWithLabels.projectId, false));
+
+        // Test that feed source shows only correct labels based on user auth
+        List<String> labelsSeenByAdmin = FeedSourceController.cleanFeedSourceLabels(feedSourceWithLabels, true).labelIds;
+        List<String> labelsSeenByViewOnlyUser = FeedSourceController.cleanFeedSourceLabels(feedSourceWithLabels, false).labelIds;
```

Comment:
It'd be nice to send off an actual HTTP request and then parse the response here. In doing things that way, we would be testing the overall API functionality and not tying things to the internal workings of the FeedSourceController.

Reply:
I agree. However, I think Landon and I agreed that the work required to mock the authorization bearers (from either the client side or the server side tests) would not be worth it at this time

Reply:
This comment is not blocking and I'm just saying it'd be a nice-to-have. However, the approach wouldn't be to mock authorization bearers. If you notice, you're already making admin-only calls to create and delete labels and you're not sending any kind of mocked bearer for that. The reason this works is because of these lines of code here:

https://github.com/ibi-group/datatools-server/blob/00b5155694ef3e36d7f480351dbf4ed9ea131564/src/main/java/com/conveyal/datatools/manager/auth/Auth0Connection.java#L55-L60

So, to test out a non-admin user, perhaps the approach would be to add some kind of query parameter instructing the `Auth0Connection#checkUser` method to create a non-admin user profile for that specific request.

Reply:
Will add this to the technical backlog document

Reply:
My #410 PR has new functionality to handle testing non-admin users, which I think might be satisfactory long term.